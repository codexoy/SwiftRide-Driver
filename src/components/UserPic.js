import React from "react"
import { getUserImage } from "../actions/utils"
import FastImage from "react-native-fast-image"

export default class UserPic extends React.Component {
	render() {
		const width = this.props.width || 34
		const img = getUserImage(this.props.user)
		return (
			<FastImage
				style={{
					...this.props.style,
					...{
						borderRadius: width / 2,
						width: width,
						height: this.props.height || 34
					}
				}}
				source={{
					uri: img
				}}
			/>
		)
	}
}
