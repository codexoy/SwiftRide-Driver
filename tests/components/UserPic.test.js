import React from "react"
import { Text } from "react-native"
import renderer from "react-test-renderer"

import UserPic from "../../src/components/UserPic"

describe("UserWithPic", () => {
	test("view renders correctly", () => {
		const user = { image: "http://example.com" }
		const tree = renderer
			.create(
				<UserPic
					style={{ width: 100 }}
					width={100}
					height={100}
					user={user}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
