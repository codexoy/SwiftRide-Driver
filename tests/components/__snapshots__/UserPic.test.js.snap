// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserWithPic view renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      },
    ]
  }
>
  <FastImageView
    resizeMode="cover"
    source={
      Object {
        "uri": "http://example.com",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
